grakn-module-sdk/src/main/java/ai/grakn/graknmodule/GraknModule.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Grakn - A Distributed Semantic Database
+ * Copyright (C) 2016  Grakn Labs Limited
+ *
+ * Grakn is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Grakn is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Grakn. If not, see <http://www.gnu.org/licenses/gpl.txt>.
+ *
+ */
+
+package ai.grakn.graknmodule;
+
+import ai.grakn.graknmodule.http.HttpEndpoint;
+import ai.grakn.graknmodule.http.HttpBeforeFilter;
+
+import java.util.List;
+
+/**
+ * An interface for author who wants to implement a Grakn module
+ *
+ * @author Ganeshwara Herawan Hananda
+ */
+public interface GraknModule {
+    String getGraknModuleName();
+
+    List<HttpBeforeFilter> getHttpBeforeFilters();
```

Comment:
java docs for these methods maybe? It will make it easier to understand how these should be implemented.

Reply:
fixed in 95eaa3d2f4e54522e5b1afea03139ec2905d2b20, 2af3d803ca214daa1eb0660c10d46396d36939a6, and 2af3d803ca214daa1eb0660c10d46396d36939a6